psa-crypto/src/types/algorithm.rs

```diff
@@ -437,6 +437,19 @@ impl AsymmetricSignature {
             }
         }
     }
+
+    /// Retrieves the specific hash that the `AsymmetricSignature` algorithm is restricted to.
+    pub fn hash(self) -> SignHash {
```

Comment:
I would personally prefer returning `Option<SignHash>` and return `None` for `EcdsaAny` and `RsaPkcs1v15SignRaw` so that all different cases are covered for this method. Maybe differentiating between those cases will be useful in the future?